socketio/src/packet.rs

```diff
@@ -234,27 +220,12 @@ impl TryFrom<&Bytes> for Packet {
 
         // data
         let json_str: String = utf8_iter.into_iter().collect();
-        let json_data: serde_json::Value = serde_json::from_str(&json_str).map_err(InvalidJson)?;
-
-        match packet.packet_type {
-            PacketId::BinaryAck | PacketId::BinaryEvent => {
-                let re_close = Regex::new(r",]$|]$").unwrap();
-                let mut str = json_data
-                    .to_string()
-                    .replace("{\"_placeholder\":true,\"num\":0}", "");
-
-                if str.starts_with('[') {
-                    str.remove(0);
-                }
-                str = re_close.replace(&str, "").to_string();
+        let json_data: Value = serde_json::from_str(&json_str).map_err(InvalidJson)?;
 
-                if str.is_empty() {
-                    packet.data = None
-                } else {
-                    packet.data = Some(str)
-                }
-            }
-            _ => packet.data = Some(json_data.to_string()),
+        packet.data = match json_data {
+            Value::Array(vec) if vec.is_empty() => None,
+            // Value::Array(vec) if vec.len() == 1 => vec.get(0).map(|v| v.to_owned()),
+            _ => Some(json_data),
```

Comment:
nit: would this still have the `_placeholder` in it?

Reply:
Yes, _placeholder will be replaced to real `Payload::Binary` in socket.rs

Reply:
packet.rs should handle all de-serialization of the packet. This will become more important if/when we want to support multiple versions.

Reply:
`BinaryEvent` can be mixed with normal data and binary data like `client.emit('test', Buffer.from([1, 2, 3]), "4", Buffer.from([5, 6]));`, so we can not just remove all `_placeholder` as we do not know where the binary data is in the sequence of received items. `socket.rs` not parse the packet, just replace `_placehoulder` to `Payload::Binary`